test/e2e/testsuites/failed_mount.go

```diff
@@ -192,6 +212,56 @@ func (t *gcsFuseCSIFailedMountTestSuite) DefineTests(driver storageframework.Tes
 		testCaseSAInsufficientAccess(specs.SkipCSIBucketAccessCheckPrefix)
 	})
 
+	testCaseSAInsufficientAccessSucceedAfterSufficientAccess := func(configPrefix string) {
+		init(configPrefix)
+		defer cleanup()
+
+		// The test driver uses config.Prefix to pass the bucket names back to the test suite.
+		bucketName := l.config.Prefix
+
+		ginkgo.By("Configuring the pod")
+		tPod := specs.NewTestPod(f.ClientSet, f.Namespace)
+		tPod.SetupVolume(l.volumeResource, volumeName, mountPath, false)
+
+		ginkgo.By("Deploying a Kubernetes service account without access to the GCS bucket")
+		saName := "sa-without-access"
+		testK8sSA := specs.NewTestKubernetesServiceAccount(f.ClientSet, f.Namespace, saName, "")
+		testK8sSA.Create(ctx)
+		tPod.SetServiceAccount(saName)
+
+		ginkgo.By("Deploying the pod")
+		tPod.Create(ctx)
+		defer tPod.Cleanup(ctx)
+
+		ginkgo.By("Checking that the pod has failed mount error PermissionDenied")
+		tPod.WaitForFailedMountError(ctx, codes.PermissionDenied.String())
+		tPod.WaitForFailedMountError(ctx, "does not have storage.objects.list access to the Google Cloud Storage bucket.")
+
+		ginkgo.By("Setting up SA IAM policy")
+		setupIAMPolicy(bucketName, f.Namespace.Name, saName)
+
+		ginkgo.By("Checking that the pod is running")
+		tPod.WaitForRunning(ctx)
```

Comment:
I am not able to fully understand how would the pod recover gracefully here for spec.SkipCSIBucketAccessCheckPrefix config. I was expecting the gcsfuse process to exit after SA check failure.

Reply:
Yes, this test should not run with the SkipCSIBucketAccessCheck volume attribute. Removed the test case.